File: eval/code_corpus/safe/command_injection.py
"""SAFE variant of command_injection.py — argv list, no shell.

Negative case: the eval expects ZERO findings for this file.
"""
import subprocess


def ping(host: str):
    # SAFE: argument vector, shell disabled — host can't break out into a command.
    return subprocess.check_output(["ping", "-c", "1", host], shell=False)

File: eval/code_corpus/safe/command_injection.js
// SAFE variant of command_injection.js — execFile with an argv array, no shell.
// Negative case: the eval expects ZERO findings for this file.

const { execFile } = require("child_process");

function ping(host, cb) {
  // SAFE: execFile passes args directly to the binary; no shell interpretation.
  execFile("ping", ["-c", "1", host], cb);
}

module.exports = { ping };
